From 70e9b6b454cfbd5130ac27fa9c15fe3b3bd5f12a Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Wed, 6 Aug 2014 03:50:14 -0400
Subject: [pci] Don't try to claim IORESOURCE_UNSET resources

Message-id: <20140806035014.1438.8511.stgit@gir.stowe>
Patchwork-id: 87466
O-Subject: [RHEL7.1 PATCH 57/69] PCI: Don't try to claim IORESOURCE_UNSET resources
Bugzilla: 1110896
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110896
Upstream Status: 29003beb7f15d3daa5a8f9afb8d007b64baa2357

commit 29003beb7f15d3daa5a8f9afb8d007b64baa2357
Author: Bjorn Helgaas <bhelgaas@google.com>
Date:   Wed Feb 26 11:25:59 2014 -0700

    PCI: Don't try to claim IORESOURCE_UNSET resources

    If the IORESOURCE_UNSET bit is set, it means we haven't assigned an address
    yet, so don't try to claim the region.

    Also, make the error messages more uniform and add info about which BAR is
    involved.

    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/setup-res.c b/drivers/pci/setup-res.c
index 7f76521..6e44313 100644
--- a/drivers/pci/setup-res.c
+++ b/drivers/pci/setup-res.c
@@ -111,18 +111,23 @@ int pci_claim_resource(struct pci_dev *dev, int resource)
  struct resource *res = &dev->resource[resource];
  struct resource *root, *conflict;
 
+ if (res->flags & IORESOURCE_UNSET) {
+  dev_info(&dev->dev, "can't claim BAR %d %pR: no address assigned\n",
+    resource, res);
+  return -EINVAL;
+ }
+
  root = pci_find_parent_resource(dev, res);
  if (!root) {
-  dev_info(&dev->dev, "no compatible bridge window for %pR\n",
-    res);
+  dev_info(&dev->dev, "can't claim BAR %d %pR: no compatible bridge window\n",
+    resource, res);
   return -EINVAL;
  }
 
  conflict = request_resource_conflict(root, res);
  if (conflict) {
-  dev_info(&dev->dev,
-    "address space collision: %pR conflicts with %s %pR\n",
-    res, conflict->name, conflict);
+  dev_info(&dev->dev, "can't claim BAR %d %pR: address conflict with %s %pR\n",
+    resource, res, conflict->name, conflict);
   return -EBUSY;
  }
 
-- 
1.7.1